From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Hide virtual devices from the nm-applet menu.
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1423819
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1440166
Forwarded: not-needed

Their presence is confusing to most users.

For one example: most virtualization systems (including LXC) set up bridges
for VMs or containers to be able to talk to the outside world. This can lead
to many extra entries in the menu, and clicking on these entries may disable
the bridge, something which users might not be able to easily recover from.

Updated: 2015-04-03

---
 src/applet.c |    8 --------
 1 file changed, 8 deletions(-)

Index: network-manager-applet-0.9.10.1/src/applet.c
===================================================================
--- network-manager-applet-0.9.10.1.orig/src/applet.c	2015-04-03 15:52:48.103587301 -0400
+++ network-manager-applet-0.9.10.1/src/applet.c	2015-04-03 16:02:39.872453669 -0400
@@ -1778,18 +1778,10 @@
 	all_devices = nm_client_get_devices (applet->nm_client);
 
 	n_items = 0;
-	n_items += add_virtual_items (NM_SETTING_BRIDGE_SETTING_NAME,
-	                              all_devices, all_connections, menu, applet);
-	n_items += add_virtual_items (NM_SETTING_BOND_SETTING_NAME,
-	                              all_devices, all_connections, menu, applet);
-	n_items += add_virtual_items (NM_SETTING_TEAM_SETTING_NAME,
-	                              all_devices, all_connections, menu, applet);
 	n_items += add_device_items  (NM_DEVICE_TYPE_ETHERNET,
 	                              all_devices, all_connections, menu, applet);
 	n_items += add_device_items  (NM_DEVICE_TYPE_INFINIBAND,
 	                              all_devices, all_connections, menu, applet);
-	n_items += add_virtual_items (NM_SETTING_VLAN_SETTING_NAME,
-	                              all_devices, all_connections, menu, applet);
 	n_items += add_device_items  (NM_DEVICE_TYPE_WIFI,
 	                              all_devices, all_connections, menu, applet);
 	n_items += add_device_items  (NM_DEVICE_TYPE_MODEM,
@@ -2735,7 +2727,11 @@
 	NMApplet *applet = NM_APPLET (user_data);
 	OfflineNotificationContextInfo *info = applet->notification_queue_data;
 	if(select_merged_notification_text (info))
-		if (!g_settings_get_boolean (applet->gsettings, PREF_DISABLE_DISCONNECTED_NOTIFICATIONS))
+		if (!g_settings_get_boolean (applet->gsettings, PREF_DISABLE_DISCONNECTED_NOTIFICATIONS)
+			&& (info->device_type != NM_DEVICE_TYPE_VLAN)
+			&& (info->device_type != NM_DEVICE_TYPE_BOND)
+			&& (info->device_type != NM_DEVICE_TYPE_TEAM)
+			&& (info->device_type != NM_DEVICE_TYPE_BRIDGE))
 			applet_do_notify (applet, info->urgency, info->title,
 			                  info->text, info->icon,
 			                  PREF_DISABLE_DISCONNECTED_NOTIFICATIONS,
@@ -2800,7 +2796,16 @@
 				info->device_type = NM_DEVICE_TYPE_MODEM;
 				info->text = g_strdup (_("Modem network"));
 			} else {
-				info->device_type = NM_DEVICE_TYPE_UNKNOWN;
+				if (NM_IS_DEVICE_VLAN (device))
+					info->device_type = NM_DEVICE_TYPE_VLAN;
+				else if (NM_IS_DEVICE_BOND (device))
+					info->device_type = NM_DEVICE_TYPE_BOND;
+				else if (NM_IS_DEVICE_TEAM (device))
+					info->device_type = NM_DEVICE_TYPE_TEAM;
+				else if (NM_IS_DEVICE_BRIDGE (device))
+					info->device_type = NM_DEVICE_TYPE_BRIDGE;
+				else
+					info->device_type = NM_DEVICE_TYPE_UNKNOWN;
 				info->text = g_strdup (_("Network"));
 			}
 
@@ -2855,7 +2860,11 @@
 	applet_common_device_state_changed (device, new_state, old_state, reason, applet);
 
 	if (   new_state == NM_DEVICE_STATE_ACTIVATED
-	    && !g_settings_get_boolean (applet->gsettings, PREF_DISABLE_CONNECTED_NOTIFICATIONS)) {
+	    && !g_settings_get_boolean (applet->gsettings, PREF_DISABLE_CONNECTED_NOTIFICATIONS)
+		&& !NM_IS_DEVICE_VLAN (device)
+		&& !NM_IS_DEVICE_BOND (device)
+		&& !NM_IS_DEVICE_TEAM (device)
+		&& !NM_IS_DEVICE_BRIDGE (device)) {
 		NMConnection *connection;
 		NMSettingConnection *s_con = NULL;
 		char *str = NULL;
